### PR TITLE
Bluetooth: samples: hci_uart_async: Use existing H4 defines

### PR DESCRIPTION
**Pull request by jhedberg**
Use existing H4 defines from hci_types.h instead of redefining them.

### CODE REVIEW

**Review by jori-nordic — DISMISSED** [review]

**Review by Thalley — DISMISSED** [review]

File: samples/bluetooth/hci_uart_async/src/hci_uart_async.c [hunk 1]
```diff
@@ -82,14 +74,14 @@ static int uart_c2h_tx(const uint8_t *data, size_t size)
 }
 
 /* Function expects that type is validated and only CMD, ISO or ACL will be used. */
-static uint32_t hci_payload_size(const uint8_t *hdr_buf, enum h4_type type)
+static uint32_t hci_payload_size(const uint8_t *hdr_buf, uint8_t type)
```
Comment (alwa-nordic) [inline samples/bluetooth/hci_uart_async/src/hci_uart_async.c:77]:
```suggestion
static uint32_t hci_payload_size(const uint8_t *hdr_buf, uint8_t h4_type)
```

The type lost the 'h4' hint. I think we should move it to the parameter name to retain self-documentation.

**Review by alwa-nordic — CHANGES_REQUESTED** [review]

File: samples/bluetooth/hci_uart_async/src/hci_uart_async.c (hunk 1 above)
Comment (jhedberg) [reply samples/bluetooth/hci_uart_async/src/hci_uart_async.c:77]: > The type lost the 'h4' hint. I think we should move it to the parameter name to retain self-documentation.

Sure. I'll ad it back.

Comment (jhedberg) [reply samples/bluetooth/hci_uart_async/src/hci_uart_async.c]: > > The type lost the 'h4' hint. I think we should move it to the parameter name to retain self-documentation.
> 
> Sure. I'll add it back.

Done.

**Review by alwa-nordic — APPROVED** [review]

**Comment by jhedberg** [comment]
@Thalley @jori-nordic could you take another look? Only difference is the type -> h4_type rename that @alwa-nordic suggested. Thanks.

**Review by Thalley — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**